app/lib/main/run_app.dart

```diff
@@ -128,14 +131,22 @@ Future<AppDependencies> initializeFlutterApp({
   // Skipping the listeners to the auth state to make the integration tests
   // work.
   //
-  // The listeners of "listenToAuthStateChanged().listen()" are not disposed.
-  // Therefore, when signing out, the app still tries to stream the Firestore
-  // documents which results into a "[cloud_firestore/permission-denied] The
-  // caller does not have permission to execute the specified operation."
-  // because the user isn't signed anymore.
+  // The code below that uses listenToAuthStateChanged creates SharezoneGateway
+  // & UserGateway which automatically subscribes to document changes in several
+  // collections in Firestore. This results in errors if the user logs out since
+  // we don't correctly dispose or close the streams inside SharezoneGateway &
+  // UserGateway. The errors will cause any integration test to automatically
+  // fail.
   //
-  // A good solution would be to refactor the listeners. But as a quick
-  // workaround we just skip the listeners for integration tests.
+  // Currently we just work around that by not executing the code below at all
+  // (and thus not creating SharezoneGateway). In the future we should ensure
+  // that the SharezoneGateway does not unnecessarily subscribe to document
+  // changes and that we can properly dispose of the subscriptions before
+  // logging out. The listeners of "listenToAuthStateChanged().listen()" are not
+  // disposed. Therefore, when signing out, the app still tries to stream the
+  // Firestore documents which results into a
+  // "[cloud_firestore/permission-denied] The caller does not have permission to
+  // execute the specified operation." because the user isn't signed anymore.
```

Comment:
Better but there is still missing that the UserGateway created inside the SharezoneApp widget tree is fine and we don't know why.     

Also the text seems a bit duplicated and confusing between the first (further up top) and this second paragraph.   

It's okay for now, if you want maybe try to make it a bit nicer but it's okay like this as well I suppose.